Fix undefined methods 'addClass' and `removeClass` on a PrototypeJS Element

markvds opened the pull request:
### Description
This problem was caused by 5a283e3ff9 and reported in #20843. The mentioned commit adds two lines in `js.phtml` with non existing methods: `panel.addClass(activePanelClass);` and `panel.removeClass(activePanelClass);`. Both methods do not exist in PrototypeJS and should have been `addClassName()` and `removeClassName()` respectively.

The incorrect method names cause a Javascript error 'Uncaught TypeError: panel.addClass is not a function' in the console and prevents the Edit Attribute form from saving.

Enabling the Swatches module, as mentioned in the issue, only solves the issue because it replaces the phtml file with its own file.

### Fixed Issues (if relevant)
1. magento/magento2#20843: Uncaught TypeError: panel.addClass is not a function when Swatches are disabled

### Manual testing scenarios
1. Disable Swatches modules (otherwise, the mentioned phtml file will not be used)
2. In admin, go to Stores > Attributes > Product
3. Click any attribute
4. Try to save the attribute
5. Nothing happens without the code change, but the attribute saves fine with the code change

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=22560) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=22560) before we can accept your contribution.<br/>

orlangur reviewed (changes requested):
@markvds please use `jQuery` equivalents instead.

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

markvds commented:
@orlangur I cannot imagine that it makes the situation any better; it's all surrounded by PrototypeJS code. I think it would even make readibility worse.

orlangur commented:
@markvds we don't need to add any new PrototypeJS-related code. Feel free to refactor any additional occurrences in file with jQuery.

barryvdh commented:
@markvds Thank you for your PR.

@orlangur I would keep the bugfix PR as minimal as possible to make it clear what the intent and actual fix of this PR is. Also to make it easier to avoid merge conflicts etc.Refactoring the prototype should happen in a seperate PR imho.

orlangur commented:
@barryvdh it will be minimal as I suggested since `jQuery` is already injected.

@markvds please apply necessary changes.

jissereitsma commented:
My 50 cents: The proposed fix of @markvds is a bugfix. However, the suggestion of @orlangur would lead to a code enhancement. They are different things. I'm not sure if postponing the bugfix is a good idea, waiting for the code enhancement. On the other hand, I'm not sure how much work the refactoring the jQuery code would be. Just my 50 cents.

orlangur commented on app/code/Magento/Catalog/view/adminhtml/templates/catalog/product/attribute/js.phtml:
```diff
@@ -45,11 +45,11 @@ function checkOptionsPanelVisibility(){
         if($('frontend_input') && ($('frontend_input').value=='select' || $('frontend_input').value=='multiselect')){
             panel.show();
-            panel.addClass(activePanelClass);
+            panel.addClassName(activePanelClass);
```
```suggestion
            jQuery(panel).addClass(activePanelClass);
```

orlangur reviewed (changes requested):
Initial code was jQuery-style. There is no need to step back. Please replace all `addClassName/removeClassName`.

orlangur reviewed (changes requested):
After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

jissereitsma commented:
@orlangur You are awesome

markvds committed:
Fix undefined methods 'addClass' and 'removeClass' on a PrototypeJS Element

The code was originally copied from the Swatches module, which already uses jQuery.
In this file however, the 'panel' variable is not a jQuery object but a DOM Element.

markvds commented:
I changed the PR. 

@orlangur I finally get your point. You basically say the method name was correct, but the context was wrong. I assumed the context was right but the method name was wrong. It's just a different view on the situation. Understanding your thoughts helps me in future PRs. Thanks for explaining!

orlangur commented:
@markvds yep, sorry for confusion, will try to formulate better next time 🤝

magento-engcom-team commented:
Hi @orlangur, thank you for the review.
[ENGCOM-5020](https://jira.corp.magento.com/browse/ENGCOM-5020) has been created to process this Pull Request

magento-engcom-team commented:
@markvds thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.
